### PR TITLE
feat: Implement modal component.

### LINKED ISSUE #203: Implement Modal component
## Usage

```
<Modal title="Change your password" isOpen={state.isOpen}>
   <Field label="Enter the new password" component={Input} type="password" />
   <Button label="Save" />
</Modal>
```
## Props
isOpen - {bool}
title - {String|Node}
size - {all the posible sizes} // @TahimiLeonBravo update this based on designs
children

### PR DESCRIPTION
**Pull request by leyanessantiago**
fixes #203

### CODE REVIEW

File: src/components/Modal/header.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        title,
+        id,
+        onClick,
+    } = props;
+    return (
+        <div className="rainbow-modal_header">
+            <ButtonIcon className="rainbow-modal_close" icon={<CloseIcon />} title="Close" onClick={onClick} />
+            <RenderIf isTrue={!!title}>
+                <header>
+                    <RenderIf isTrue={!!title}>
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/header.js]: why two RenderIf?

File: src/components/Modal/headerTitle.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import React from 'react';
+import PropTypes from 'prop-types';
+import './styles.css';
+
+export default function HeaderTitle({ id, title }) {
+    if (typeof title === 'string') {
+        return (
+            <h2
+                id={id}
+                className="">
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/headerTitle.js]: className="" ?

File: src/components/Modal/index.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        const {
+            title,
+            style,
+            children,
+            footer,
+            isOpen,
+            onRequestClose,
+        } = this.props;
+        return (
+            createPortal(
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/index.js]: in this same file create a ModalComponent with all this. And then use:
```
export default function Modal(props) {
    return createPortal(ModalComponent);
}
```

File: src/components/Modal/index.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                <RenderIf isTrue={isOpen}>
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/index.js]: don't use RenderIf here

File: src/components/Modal/index.js [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+                    <div className="rainbow-backdrop rainbow-modal--fade-in">
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/index.js]: rainbow-backdrop don't follow the className convention and also the this.getContainerClassNames() should be here and not in section element. Remember that className prop must be applied to container element.

File: src/components/Modal/index.js [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+                        <section
+                            role="dialog"
+                            tabIndex="-1"
+                            aria-labelledby={this.modalHeadingId}
+                            aria-modal="true"
+                            aria-hidden={!isOpen}
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/index.js]: not needed only aria-modal

File: src/components/Modal/index.js [hunk 7]
```diff
@@ -0,0 +1,113 @@
+import React, { Component } from 'react';
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/index.js]: the focus should stay in and cycle through the modal’s content and not to underlying page

File: src/components/Modal/index.js [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+                    </div>
+                </RenderIf>,
+                document.body,
+            )
+        );
+    }
+}
+
+Modal.propTypes = {
+    isOpen: PropTypes.bool,
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/index.js]: add description

File: src/components/Modal/index.js [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    /** The title . */
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/index.js]: copy the same of the card

File: src/components/Modal/index.js [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * This prop that should not be visible in the documentation.
+     * @ignore
+     */
+    children: PropTypes.node,
+    /** A CSS class for the outer element, in addition to the component's base classes. */
+    className: PropTypes.string,
+    /** An object with custom style applied to the outer element. */
+    style: PropTypes.object,
+    footer: PropTypes.node,
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/index.js]: add description

File: src/components/Modal/styles.scss [hunk 11]
```diff
@@ -0,0 +1,143 @@
+@import './../../styles/colors.scss';
+@import './../../styles/shadows.scss';
+@import './../../styles/border-radius.scss';
+
+.rainbow-backdrop {
+    //transition: opacity 0.4s linear;
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/styles.scss]: remove comments

File: src/components/Modal/styles.scss [hunk 12]
```diff
@@ -0,0 +1,143 @@
...
+    width: 100%;
+    height: 100%;
+    //opacity: 0;
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/styles.scss]: same ^^

File: src/components/Modal/styles.scss [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+    left: 0;
+    background: rgba(43, 40, 38, 0.6);
+    z-index: 90000;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+}
+
+.rainbow-modal {
+    //opacity: 0;
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/styles.scss]: same ^^

File: src/components/Modal/styles.scss [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+    visibility: visible;
+    //transition: transform 0.1s linear, opacity 0.1s linear;
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/styles.scss]: same ^^

File: src/components/Modal/styles.scss [hunk 15]
```diff
@@ -0,0 +1,143 @@
...
+    position: relative;
+    z-index: 90001;
+    width: 50%;
+    max-width: none;
+    min-width: 40rem;
+}
+
+.rainbow-modal_container {
+    position: relative;
+    //transform: translate(0, 0);
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/styles.scss]: same ^^

File: src/components/Modal/styles.scss [hunk 16]
```diff
@@ -0,0 +1,143 @@
...
+    //transition: transform 0.1s linear, opacity 0.1s linear;
```
Comment (LeandroTorresSicilia) [inline src/components/Modal/styles.scss]: same ^^

File: src/components/Modal/styles.scss [hunk 17]
```diff
@@ -0,0 +1,151 @@
...
+    max-width: none;
+    min-width: 20rem;
+
+    &:focus {
+        outline: 0;
+    }
+}
+
+.rainbow-modal_container {
+    position: relative;
```
Comment (TahimiLeonBravo) [inline src/components/Modal/styles.scss]: You can use a mixin for position styles, this is repeat in two class

**Pull request merged**